src/main/java/com/ibm/eventstreams/connect/mqsource/processor/JmsToKafkaHeaderConverter.java

```diff
@@ -0,0 +1,55 @@
+package com.ibm.eventstreams.connect.mqsource.processor;
+
+import org.apache.kafka.connect.header.ConnectHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+import java.util.ArrayList;
+import java.util.Collections;
+
+/*
+ * Single responsibility class to copy JMS properties to Kafka headers.
+ *
+ * */
+public class JmsToKafkaHeaderConverter {
+
+	private static final Logger log = LoggerFactory.getLogger(JmsToKafkaHeaderConverter.class);
+
+	/**
+	 * Copies the JMS properties to Kafka headers.
+	 *
+	 * @param message        JMS message.
+	 * @param messageBodyJms flag whether incoming message is treated as jms message.
+	 * @return Kafka connect headers.
+	 */
+	public ConnectHeaders convertJmsPropertiesToKafkaHeaders(boolean messageBodyJms, Message message) {
```

Comment:
Change to a single parameter - the message - and only call if messageBodyJms == true.l

Reply:
@AndrewJSchofield Changed as per the suggestion.